querydb/src/main/scala/io/joern/scanners/kotlin/PathTraversals.scala

```diff
@@ -0,0 +1,124 @@
+package io.joern.scanners.kotlin
+
+import io.joern.scanners._
+import io.joern.console._
+import io.joern.dataflowengineoss.queryengine.EngineContext
+import io.joern.dataflowengineoss.semanticsloader.Semantics
+import io.joern.dataflowengineoss.language._
+import io.joern.macros.QueryMacros._
+import io.shiftleft.semanticcpg.language._
+
+object PathTraversals extends QueryBundle {
+  implicit val engineContext: EngineContext = EngineContext(Semantics.empty)
+  implicit val resolver: ICallResolver      = NoResolve
+
+  @q
+  def unzipDirectoryTraversal(): Query =
+    Query.make(
+      name = "unzip-directory-traversal",
+      author = Crew.claudiu,
+      title = "Zip entries not checked before unzipping",
+      description = "-",
+      score = 4,
+      withStrRep({ cpg =>
+        val zipEntryTypeFullNames =
+          List("org.apache.commons.compress.archivers.ArchiveEntry", "java.util.zip.ZipEntry")
+        def zipEntryParams =
+          cpg.method.parameter.typeFullNameExact(zipEntryTypeFullNames: _*)
+        def fileOutStreamInit =
+          cpg.method.fullNameExact("java.io.FileOutputStream.<init>:void(java.io.File)").callIn
+        def pathStartsWithCalls =
+          cpg.method.fullNameExact("java.nio.file.Path.startsWith:boolean(java.nio.file.Path)").callIn
+
+        def uncheckedZipEntryParameters = zipEntryParams.filterNot { param =>
+          pathStartsWithCalls.argument.reachableByFlows(param).nonEmpty
```

Comment:
```suggestion
          pathStartsWithCalls.argument(0).reachableByFlows(param).nonEmpty
```

(Not tested - but I think this check should include the argument index.)

Reply:
yess, good catch, thx